library/alloc/src/string.rs

```diff
@@ -1022,8 +1023,9 @@ impl String {
     #[inline]
     #[must_use]
     #[stable(feature = "string_as_str", since = "1.7.0")]
-    pub fn as_str(&self) -> &str {
-        self
+    #[rustc_const_unstable(feature = "const_vec_string_slice", issue = "129041")]
+    pub const fn as_str(&self) -> &str {
+        unsafe { str::from_utf8_unchecked(self.vec.as_slice()) }
```

Comment:
Why doesn't tidy require a `SAFETY` Comment Herr?

Reply:
@mammothbane please add a `// SAFETY: ...` comment here (it's pretty obvious and I know this was just moved from another location, but ideally we want to have those comments everywhere)